File: 0798/798.cs
public class Solution {
    public int SmallestRotationWithHighestSco(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(SmallestRotationWithHighestSco(root.left), SmallestRotationWithHighestSco(root.right));
    }
}

File: 0798/798.go
func smallestRotationWithHighestSco(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := smallestRotationWithHighestSco(root.Left)
	r := smallestRotationWithHighestSco(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}